src/Services/CarePlanService.php

```diff
@@ -0,0 +1,276 @@
+
+        (new UuidRegistry(['table_name' => self::PATIENT_TABLE]))->createMissingUuids();
+        (new UuidRegistry(['table_name' => self::ENCOUNTER_TABLE]))->createMissingUuids();
+
+        parent::__construct(self::CARE_PLAN_TABLE);
+        $this->codeTypesService = new CodeTypesService();
+    }
+
+    public function getOne($uuid, $puuid = null)
+    {
+        $search = [
+            'uuid' => new TokenSearchField('uuid', [new TokenSearchValue($uuid, null, false)])
+        ];
+        if (isset($puuid)) {
+            $search['puuid'] = new ReferenceSearchField('puuid', [new ReferenceSearchValue($puuid, 'Patient', true)]);
+        }
+        return $this->search($search);
+    }
+
+    /**
+     * Returns a list of all care plan resources.  Search array can be a simple key => value array which does an exact
+     * match on passed in value.  For more complicated searching @see CarePlanService::search().
+     * @param $search a key => value array
+     * @param bool $isAndCondition Whether the search should be a UNION of search values or INTERSECTION of search values
+     * @return ProcessingResult
+     */
+    public function getAll($search, $isAndCondition = true)
+    {
+        $newSearch = [];
+        foreach ($search as $key => $value) {
+            if (!$value instanceof ISearchField) {
+                $newSearch[] = new StringSearchField($key, [$value], SearchModifier::EXACT);
+            } else {
+                $newSearch[$key] = $value;
+            }
+        }
+        return $this->search($search, $isAndCondition);
+    }
+
+    public function search($search, $isAndCondition = true)
+    {
+        if (isset($search['uuid']) && $search['uuid'] instanceof ISearchField) {
+            $this->populateSurrogateSearchFieldsForUUID($search['uuid'], $search);
+        }
+        // this value is defined in code so we don't need to db escape it.
+        $carePlanType = $this->carePlanType;
+        $planCategory = "assess-plan";
+        if ($carePlanType === self::TYPE_GOAL) {
+            $planCategory = "goal";
+        }
+        $sql = "SELECT
+                patients.puuid
+                ,patients.pid
+                ,encounters.euuid
+                ,encounters.eid
+                ,fcp.form_id
+                ,fcp.code
+                ,fcp.codetext
+                ,fcp.description
+                ,fcp.date
+                ,l.`notes` AS moodCode
+                ,category.careplan_category
+                 FROM 
+                 (
+                    select 
+                        id AS form_id
+                        ,code
+                        ,codetext
+                        ,description
+                        ,`date`
+                        ,`encounter`
+                        ,`pid`
+                        ,`care_plan_type`
+                    FROM 
+                        form_care_plan
+                    WHERE
+                        `care_plan_type` = '$carePlanType'
```

Comment:
prob still better to bind this even though already whitelisted in case anybody ever decides to expand this or decide to allow an item with a ? character in it

Reply:
actually, after thinking about this, can't see why need to bind this, so can disregard above post :)